### PR TITLE
automatic file transfer & unit tests

### PR DESCRIPTION
**Pull request by patcmorneau**
these these fails on purpose

files_websocket fail because :
```
        TEST(FileWebsocket, deleteFile) {
	        ASSERT_FALSE(true) << " no tests for deleteFile()";
        }
``` 
setting_msg/tests/test_zero_imu_offset_service.cpp fail because

line 136 : 
`ASSERT_FALSE(true) << "The test needs to be verified by a math expert but should be good";`

### CODE REVIEW

File: install/sync_logfiles.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-USER="Posseidon"
-FILES_LOCATION="~/Poseidon/www/webroot/record"
-PASSWORD_LESS="poseidon"
-KEEP_ORIGINAL_FILES=true
+
+DESTINATION_PATH="destination_Path"
+SERVER="hostname or ip"
+USER="HB_XX"
+FILES_LOCATION="/home/ubuntu/Poseidon/www/webroot/record/" # without the last '/' it will transfer the folder instead of files
+PASSWORD_LESS="test"
```
Comment (glabmoris) [inline install/sync_logfiles.sh:8]: Why _LESS ?

File: src/workspace/src/hydroball_config_websocket/CMakeLists.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@ project(hydroball_config_websocket)
 
 ## Compile as C++11, supported in ROS Kinetic and newer
 # add_compile_options(-std=c++11)
-add_definitions(-std=c++11 -Wall -g)
+add_definitions(-std=c++2a -Wall -g)
```
Comment (glabmoris) [inline src/workspace/src/hydroball_config_websocket/CMakeLists.txt:10]: Why do we need c++20 ?

File: src/workspace/src/hydroball_config_websocket/tests/test_hydroball_config_websocket.cpp [hunk 3]
```diff
@@ -1,38 +1,112 @@
+/*
```
Comment (glabmoris) [inline src/workspace/src/hydroball_config_websocket/tests/test_hydroball_config_websocket.cpp:1]: Why is the test licensed to Peter Thorson?

**Review by glabmoris — CHANGES_REQUESTED** [review]

File: src/workspace/src/hydroball_config_websocket/tests/test_hydroball_config_websocket.cpp (hunk 3 above)
Comment (patcmorneau) [reply src/workspace/src/hydroball_config_websocket/tests/test_hydroball_config_websocket.cpp:1]: 80% - 90%  comes from here https://github.com/zaphoyd/websocketpp/blob/master/tutorials/utility_client/step6.cpp

File: install/sync_logfiles.sh (hunk 1 above)
Comment (patcmorneau) [reply install/sync_logfiles.sh:8]: could change the name of the variable to rsa_key

File: src/workspace/src/setting_msg/tests/test_zero_imu_offset_service.cpp [hunk 4]
```diff
@@ -97,37 +97,43 @@ TEST(ZeroImuOffsetTestSuite, testZeroImuOffset) {
...
+	testPitch -= expectedPitchOffset;
+	testRoll -= expectedRollOffset;
+	
     tf2::Quaternion q;
-    q.setRPY(D2R(testRoll),D2R(testPitch),D2R(testHeading));
+    q.setRPY(D2R(testRoll),D2R(testPitch),D2R(expectedHeadingOffset));
     imuMsg.orientation = tf2::toMsg(q);
     imuPublisher.publish(imuMsg);
 
     sleep(2); // give enough time for StateController's state to be updated
```
[the remark this answers is not in the document]
Comment (patcmorneau) [reply src/workspace/src/setting_msg/tests/test_zero_imu_offset_service.cpp:122]: the way the test was built before, there was no way for it to pass... didn't change the architecture of the test, I tried to make a relevant test for the package.

File: src/workspace/src/setting_msg/tests/test_zero_imu_offset_service.cpp [hunk 5]
```diff
@@ -119,7 +119,7 @@ TEST(ZeroImuOffsetTestSuite, testZeroImuOffset) {
     imuMsg.orientation = tf2::toMsg(q);
     imuPublisher.publish(imuMsg);
 
-    sleep(2); // give enough time for StateController's state to be updated
+    //sleep(2); // give enough time for StateController's state to be updated
```
Comment (glabmoris) [inline src/workspace/src/setting_msg/tests/test_zero_imu_offset_service.cpp:122]: This needs to be double checked

**Review by glabmoris — CHANGES_REQUESTED** [review]

File: install/sync_logfiles.sh [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
-USER="Posseidon"
-FILES_LOCATION="~/Poseidon/www/webroot/record"
-PASSWORD_LESS="poseidon"
-KEEP_ORIGINAL_FILES=true
+
+DESTINATION_PATH="destination_Path"
+SERVER="hostname or ip"
+USER="HB_XX"
+FILES_LOCATION="/home/ubuntu/Poseidon/www/webroot/record/" # without the last '/' it will transfer the folder instead of files
+PRIVATE_RSA_KEY="test"
```
Comment (glabmoris) [inline install/sync_logfiles.sh:8]: Make sure you also change where it's used

**Review by glabmoris — CHANGES_REQUESTED** [review]

File: src/workspace/src/hydroball_config_websocket/CMakeLists.txt (hunk 2 above)
Comment (glabmoris) [reply src/workspace/src/hydroball_config_websocket/CMakeLists.txt:10]: Can't we just use 
testConfiguration.find(key) != testConfiguration.end() ?

**Pull request merged**